src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,163 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        switch (size) {
+            case 0 :
+                first = new Node<>(null, value, null);
+                last = first;
+                break;
+            case 1 :
+                last = new Node<>(first, value, null);
+                first.next = last;
+                break;
+            default :
+                last.next = new Node<>(last, value, null);
+                last = last.next;
+        }
+        size++;
```

Comment:
```suggestion
    Node<T> newNode = new Node<>(last, value, null);
    if (size == 0) {
      first = newNode;
    } else {
      last.next = newNode;
    }
    last = newNode;
    size++;
```
I don't think we need `case 1` at all, so we can simplify this